test/WormholeL1VotePool.t.sol

```diff
@@ -259,7 +292,81 @@ contract _ReceiveCastVoteWormholeMessages is L1VotePoolTest {
       new bytes[](0),
       bytes32(uint256(uint160(address(l2VoteAggregator)))),
       L2_CHAIN.wormholeChainId,
-      bytes32("")
+      bytes32(""),
+      false
+    );
+  }
+
+  function testFuzz_RevertWhen_VoteBeforeProposalStart(
+    uint32 _l2Against,
+    uint32 _l2InFavor,
+    uint32 _l2Abstain,
+    uint32 _l2NewAgainst,
+    uint32 _l2NewInFavor,
+    uint32 _l2NewAbstain
+  ) public {
+    vm.assume(_l2NewAgainst < _l2Against);
+    vm.assume(_l2NewInFavor < _l2InFavor);
+    vm.assume(_l2NewAbstain < _l2Abstain);
+
+    vm.selectFork(targetFork);
+
+    l1Erc20.approve(
+      address(l1VotePool), uint96(_l2Against) + uint96(_l2InFavor) + uint96(_l2Abstain)
+    );
+    l1Erc20.mint(address(this), uint96(_l2Against) + uint96(_l2InFavor) + uint96(_l2Abstain));
+    l1Erc20.delegate(address(l1VotePool));
+
+    uint256 _proposalId = l1VotePool.createProposalVote(address(l1Erc20));
+    console2.logString("vote created");
+
+    vm.prank(L1_CHAIN.wormholeRelayer);
+    vm.expectRevert("Governor: vote not currently active");
```

Comment:
Is the logic in this test that when we call `createProposalVote` we create a new proposal and it's start time is guaranteed to be before the vote start, and that's why we don't roll to test `RevertWhen_VoteBeforeProposalStart`?

Reply:
Yes, the proposal isn't active until a grace period is met, and our mock Governor has a grace period of 4 blocks. Since we create a proposal in the same block we test it will never be past the grace period.